### ISSUE TITLE
netkvm: Increase NumRssQueues Maximum to Support More CPU Cores

### ISSUE THREAD

**Issue by zjmletang** [virtio-win/kvm-guest-drivers-windows#1227]
I’ve noticed that for netkvm, the maximum value of NumRssQueues in the inf file is currently set to 32. I would like to ask if this value can be increased, perhaps to 128 or similar. This is because more and more high-performance machines now come with a significantly higher number of CPU cores.

**Comment by YanVugenfirer** [comment]
I agree that 32 might be too low at the current days, but the problem is the certification. If we declare 128 or more, we have to certify on a system with 128 CPUs.

My suggestion is to make it vendor-specific in value and use the existing mechanism for vendor customization in the INF.

@ybendito what do you think?

**Comment by YanVugenfirer** [comment]
There is also #define PARANDIS_RSS_MAX_RECEIVE_QUEUES (32) in ndis56common.h

**Comment by ybendito** [comment]
@zjmletang I suggest to provide a PR making this vendor-configurable. Do you want to develop one?

**Comment by zjmletang** [comment]
@ybendito ，I’m definitely interested in developing a PR to make this vendor-configurable. I'll start working on it later

**Comment by zjmletang** [comment]
An update after a long silence: in the year and a half since this issue, I've had exactly one request for more than 32 RSS queues. The need is real but clearly low-priority, so I've dropped the vendor-specific idea on my side.

The good news from my verification with the current HLK kit: 1c_Mini6RSSOids is the only test I've seen that requires the system to have enough RSS-capable CPUs, and what it checks is the effective *NumRssQueues value — not the INF max. With the default kept at 16, it passes on a 16-vCPU system , regardless of what the INF max declares. And in the latest HLK filter update this test is even on the exemption list — test VMs that don't meet the CPU requirement just get it filtered. So with the current kit, the certification system size doesn't really constrain the INF max.

Given that, whenever the demand justifies it, I feel a plain upstream bump of the INF max would be the simplest path. The change surface looks small: the INF max, PARANDIS_RSS_MAX_RECEIVE_QUEUES (which also sizes the ReceiveQueues array), and one detail worth noting: RSSMaxQueuesNumber is a CCHAR, so 128 wouldn't fit as-is — 64 would (and 64 happens to match the per-processor-group CPU limit on x64 anyway). Curious to hear your thoughts.

**Comment by YanVugenfirer** [comment]
@kostyanf14 @ybendito 

I am OK with bumping to 64.

**Comment by kostyanf14** [comment]
@YanVugenfirer I can confirm that HLK does not care about RSS = CPU any more. We run HLK matrix with 4/8 CPU and all tests are passed. I agree with bump too.
